src/instructlab/lab.py

```diff
@@ -712,15 +712,36 @@ def chat(
 def download(ctx, repository, release, filename, model_dir):
     """Download the model(s) to train"""
     click.echo(f"Downloading model from {repository}@{release} to {model_dir}...")
+    if (
+        "HF_TOKEN" not in os.environ
+        and repository != "instructlab/merlinite-7b-lab-GGUF"
+    ):
+        raise ValueError(
+            "HF_TOKEN var needs to be set in your environment to download HF Model. The HF Token is used to authenticate your identity to the Hugginface Hub."
+        )
     try:
         if ctx.obj is not None:
             hf_logging.set_verbosity(ctx.obj.config.general.log_level.upper())
-        hf_hub_download(
-            repo_id=repository,
-            revision=release,
-            filename=filename,
-            local_dir=model_dir,
-        )
+        files = list_repo_files(repo_id=repository, token=os.getenv("HF_TOKEN"))
```

Comment:
@cdoern why not download the entire repository via `snapshot_download()` instead of going file by file? https://huggingface.co/docs/huggingface_hub/en/guides/download#download-an-entire-repository

Reply:
This looks like a good idea to me +1

Reply:
Apologies @alimaredia -- once you posted this review, that made mergify happy that you looked at it.  If you had done "request changes" it would have blocked it, just FYI.

I probably should just set a `hold` label and let you remove it